mne/io/eeglab/eeglab.py

```diff
@@ -226,6 +233,12 @@ class RawEEGLAB(_BaseRaw):
         Names or indices of channels that should be designated
         EOG channels. If 'auto', the channel names containing
         ``EOG`` or ``EYE`` are used. Defaults to empty tuple.
+    event_id : dict
+        By default, events are read from the input file by dropping every
+        non-integer part of events containing integers, and completely
+        dropping any events without integer parts. If non-integer events
+        should be read, this should be a dict mapping from their names to
+        integers, e.g. dict(fmri_scan_onset=199, recording_start=255).
```

Comment:
>okay, how about we allow event_id. If it's None, use hashes. But otherwise, the user can write two lines of code to get the least surprising solution they want by passing it as event_id.

The problem with this is that it's a lot of work upfront you are not forced to spend with any other package. Imagine I have a very typical data set with 50 or 75 items of the form "S1", "S2", ... "S10", ... . (I have 5 data sets precisely like this I am currently working with, from 4 different people.)
As-is, the reader returns events 1, 2, 10, ..., as I think everyone would expect it to return (based on for example this being precisely how EGI or BV readers, including the EEGLAB ones and our own, work), and e.g. new EEGLAB people can immediately work with that.

Imagine you're a new user and trying to access your old EEGLAB files and it returned hashes; super strange! No other software returns hashes. 

Imagine it drops *everything* unless you first specify it in a very long (50 entry) event_id dict. You'll probably think this is a stupid software and look for something more automated.

Maybe I'm too optimistic here, but to me,
```
event_id.get(t, int("".join([x for x in t if x.isdigit()])))
```
doesn't seem like particularly failure-prone code.

The only problem I see is it will for example make "S1" and "R1" look the same. I think there probably should be a warning when this is happening.